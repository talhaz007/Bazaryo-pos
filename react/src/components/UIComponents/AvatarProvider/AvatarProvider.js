import React from "react";
import Avatar from "../Avatar/Avatar";
import "./AvatarProvider.scss";

const AvatarProvider = (props) => {
    const {
        providerImage,
        providerName,
        className,
        barClassName,
        onClickChangePhoto,
        onClickMain,
    } = props;
    return (
        <div className={`provider-avatar-wrapper ${className || ""}`}>
            <Avatar
                userImage={providerImage}
                userName={providerName}
                size="large"
                className="provider-avatar"
                onClickChangePhoto={onClickChangePhoto}
                onClickMain={onClickMain}
            />
            <div className={`provider-bar ${barClassName || ""}`}>
                <span>Provider</span>
            </div>
        </div>
    );
};

export default AvatarProvider;
