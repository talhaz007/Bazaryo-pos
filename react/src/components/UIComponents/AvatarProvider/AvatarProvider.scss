$background-bar-color: #ec5252;
$font-bar-color: #fff;
.provider-avatar-wrapper {
    position: relative;
    width: 128px;
    height: 128px;
    display: flex;
    align-items: center;
    justify-content: center;
    .provider-bar {
        position: absolute;
        text-align: center;
        text-transform: uppercase;
        font-size: 14px;
        line-height: 24px;
        background: $background-bar-color;
        border-radius: 4px;
        width: 92px;
        color: $font-bar-color;
        bottom: 0;
        display: none;
        span {
            width: 100%;
        }
    }
}
