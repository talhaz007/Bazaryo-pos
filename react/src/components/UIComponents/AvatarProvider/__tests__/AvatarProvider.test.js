import React from 'react';
import {shallow} from 'enzyme';
import {describe, it, expect} from '@jest/globals';
import AvatarProvider from '../AvatarProvider';

describe('AvatarProvider', () => {
    let wrapper;

    it('should be rendered', () => {
        wrapper = shallow(<AvatarProvider/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('shows the correct text in the bar for provider', () => {
        wrapper = shallow(<AvatarProvider/>);
        const text = wrapper.find('div div span').text();
        expect(text).toEqual('Provider');
    });
});
