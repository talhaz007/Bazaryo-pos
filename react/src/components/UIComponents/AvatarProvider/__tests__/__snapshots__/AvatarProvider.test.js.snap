// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarProvider should be rendered 1`] = `
<div
  className="provider-avatar-wrapper "
>
  <Avatar
    className="provider-avatar"
    size="large"
  />
  <div
    className="provider-bar "
  >
    <span>
      Provider
    </span>
  </div>
</div>
`;
